realsense_ros_camera/src/realsense_node_factory.cpp

```diff
@@ -79,6 +79,49 @@ void RealSenseNodeFactory::onInit()
             exit(1);
         }
 
+        // we found the device. Restart the device and wait for it to come up again
```

Comment:
Is there anything we can check to see if a device is in a bad state before blindly doing a hardware reset? With the RS200, there were failures that could be detected that indicated it needed to be reset. Avoiding a mandatory 5 second delay at startup if it's not actually needed would be nice.

Reply:
Not quite sure on that one. The failures can occur when trying to setup the profile, stream, or trying to retrieve the frame. Based on information I found on github once the device was reset it cannot be used as the pointer is invalid. Thus we would have to restart the nodelet to build the whole software stack from scratch for the camera. This is a rough workaround to give is breathing room to develop something more robust and better.